UMLEditor/UMLEditor/Views/MainWindow.axaml.cs

```diff
@@ -535,7 +534,7 @@ private void Add_Field_OnClick(object sender, RoutedEventArgs e)
                                 currentClass.AddField(fieldType,targetField);
```

Comment:
The (controller) is encroaching on model territory. The operation of the list of classes and the list of relationships (and any objects stored within them) should be done exclusively within the Diagram (who is in model land). 

- GetClassByName needs to be made private in Diagram
- Any view code that was relying on that beforehand and would break because of the change to the access modifier needs to be reconsidered / have a function in Diagram that does the equivalent action. 
- Validation of class name, field name, field type should happen on the Diagram. Diagram should throw (or let escape) any exceptions from this, the caller should have a generic try / catch to catch any such exceptions. That should reduce the size of these event handlers.

Reply:
Marking this as resolved as we are going to merge this and make a new refactor branch to deal with the above issues.